src/kagglehub/models_helpers.py

```
@@ -0,0 +1,227 @@
+    return string
+
+
+class DirectoryArchive:
+    def __init__(self, fullpath, format):  # noqa: A002
+        self._fullpath = fullpath
+        self._format = format
+        self.name = None
+        self.path = None
+
+    def __enter__(self):
+        self._temp_dir = tempfile.mkdtemp()
+        _, dir_name = os.path.split(self._fullpath)
+        self.path = shutil.make_archive(os.path.join(self._temp_dir, dir_name), self._format, self._fullpath)
+        _, self.name = os.path.split(self.path)
+        return self
+
+    def __exit__(self, *args):
+        shutil.rmtree(self._temp_dir)
+
+
+class File(object):  # noqa: UP004
+    def __init__(self, init_dict):
+        parsed_dict = {k: parse(v) for k, v in init_dict.items()}
+        self.__dict__.update(parsed_dict)
+        self.size = File.get_size(self.totalBytes)
+
+    def __repr__(self):
+        return self.ref
+
+    @staticmethod
+    def get_size(size, precision=0):
+        suffixes = ["B", "KB", "MB", "GB", "TB"]
+        suffix_index = 0
+        while size >= 1024 and suffix_index < 4:  # noqa: PLR2004
+            suffix_index += 1
+            size /= 1024.0
+        return "%.*f%s" % (precision, size, suffixes[suffix_index])
+
+
+def _upload_blob(file_path):
+    data = {
+        "type": "model",
+        "name": os.path.basename(file_path),
+        "contentLength": os.path.getsize(file_path),
+        "lastModifiedEpochSeconds": int(os.path.getmtime(file_path)),
+    }
+    api_client = KaggleApiV1Client()
+    response = api_client.post("/blobs/upload", data=data)
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    with open(file_path, "rb") as f:
+        file_data = f.read()
+
+    headers = {"Content-Type": "application/octet-stream"}
+    try:
+        requests.put(response["createUrl"], data=file_data, headers=headers, timeout=10)
+    except requests.exceptions.Timeout:
+        # Handle the timeout exception
```

Probably fine to let the Timeout exception bubble up. Also, uploading large file may take a lot of time. Is the timeout in second here?

Many upload will take several minutes.

---

i think there is a default timeout of 600 seconds (10 mins). removed anyway since its fine to let it bubble up